nzedb/processing/Videos.php

```diff
@@ -84,15 +84,283 @@ protected function getSiteIDFromVideoID($siteColumn, $videoID)
+	 *
+	 * @param        $title
+	 * @param        $type
+	 * @param int    $source
+	 *
+	 * @return array|bool
+	 */
+	public function getByTitleLikeQuery($title, $type, $source = 0)
+	{
+		$return = false;
+
+		if ($title) {
+			$return = $this->pdo->queryOneRow(
+				sprintf("
+					SELECT v.id
+					FROM videos v
+					LEFT JOIN videos_akas va ON v.id = va.videos_id
+					WHERE (v.title %1\$s
+					OR va.title %1\$s)
+					AND type = %2\$d %3\$s",
+					$this->pdo->likeString(rtrim($title, '%'), false, false),
+					$type,
+					($source > 0 ? 'AND v.source = ' . $source : '')
+				)
+			);
+		}
+		return $return;
+	}
+
+	/**
+	 * Inserts aliases for videos
+	 *
+	 * @param       $videoId
+	 * @param array $aliasArr
+	 */
+	public function addAliases($videoId, $aliasArr = array())
+	{
+		if (!empty($aliasArr) && $videoId > 0) {
+			foreach ($aliasArr AS $key => $title) {
+				// Check if we have the AKA already
+				$check = $this->getAliases(0, $title);
+
+				if ($check === false) {
+					$this->pdo->queryInsert(
+						sprintf('
+							INSERT INTO videos_akas
+							(videos_id, title)
+							VALUES (%d, %s)',
+							$videoId,
+							$this->pdo->escapeString($title)
+						)
+					);
+				}
+			}
+		}
+	}
+
+	/**
+	 * Retrieves all aliases for given VideoID or VideoID for a given alias
+	 *
+	 * @param int    $videoId
+	 * @param string $alias
+	 *
+	 * @return bool|\PDOStatement
+	 */
+	public function getAliases($videoId = 0, $alias = '')
+	{
+		$return = false;
+		$sql = '';
+
+		if ($videoId > 0) {
+			$sql = 'videos_id = ' . $videoId;
+		} else if ($alias !== '') {
+			$sql = 'title = ' . $this->pdo->escapeString($alias);
+		}
+
+		if ($sql !== '') {
+			$return = $this->pdo->query('
+				SELECT *
+				FROM videos_akas
+				WHERE ' . $sql, true
+			);
+		}
+		return (empty($return) ? false : $return);
+	}
+
+	/**
+	 * This function turns a roman numeral into an integer
+	 *
+	 * @param string $string
+	 *
+	 * @return int $e
+	 */
+	public function convertRomanToInt($string) {
+		switch ($string) {
+			case 'i': $e = 1;
+				break;
+			case 'ii': $e = 2;
+				break;
+			case 'iii': $e = 3;
+				break;
+			case 'iv': $e = 4;
+				break;
+			case 'v': $e = 5;
+				break;
+			case 'vi': $e = 6;
+				break;
+			case 'vii': $e = 7;
+				break;
+			case 'viii': $e = 8;
+				break;
+			case 'ix': $e = 9;
+				break;
+			case 'x': $e = 10;
+				break;
+			case 'xi': $e = 11;
+				break;
+			case 'xii': $e = 12;
+				break;
+			case 'xiii': $e = 13;
+				break;
+			case 'xiv': $e = 14;
+				break;
+			case 'xv': $e = 15;
+				break;
+			case 'xvi': $e = 16;
+				break;
+			case 'xvii': $e = 17;
+				break;
+			case 'xviii': $e = 18;
+				break;
+			case 'xix': $e = 19;
+				break;
+			case 'xx': $e = 20;
+				break;
+			default:
+				$e = 0;
+		}
+		return $e;
+	}
+
+	/**
+	 * Get a country code for a country name.
+	 *
+	 * @param string $country
+	 *
+	 * @return mixed
+	 */
+	public function countryCode($country)
```

Comment:
Should also be moved to somewhere more appropriate. nzedb\Utils\Country class may be.